common/djangoapps/header_control/middleware.py

```diff
@@ -0,0 +1,34 @@
+"""
+Middleware used for adjusting headers in a response before it is sent to the end user.
+"""
+
+
+class HeaderControlMiddleware(object):
+    """
+    Middleware that can modify/remove headers in a response.
+
+    This can be used, for example, to remove headers i.e. drop any Vary headers to improve cache performance.
+    """
+
+    def process_response(self, _request, response):
+        """
+        Processes the given response, potentially remove or modifying headers.
+        """
+
+        if len(getattr(response, 'remove_headers', [])) > 0:
```

Comment:
nit: it's not necessary to call `len` here. An empty list is truthy. Same goes for L25. It's not necessary to change this, though, this is fine for this pr!